rasa/utils/train_utils.py

```diff
@@ -58,14 +80,35 @@ def update_similarity_type(config: Dict[Text, Any]) -> Dict[Text, Any]:
     Returns: updated model configuration
     """
     if config.get(SIMILARITY_TYPE) == AUTO:
-        if config[LOSS_TYPE] == SOFTMAX:
+        if config[LOSS_TYPE] == CROSS_ENTROPY:
             config[SIMILARITY_TYPE] = INNER
         elif config[LOSS_TYPE] == MARGIN:
             config[SIMILARITY_TYPE] = COSINE
 
     return config
 
 
+def update_loss_type(config: Dict[Text, Any]) -> Dict[Text, Any]:
```

Comment:
we already have a function for deprecating warnings: `check_core_deprecated_options`, `check_deprecated_options`. I'd suggest to reuse them

Reply:
Those functions raise a warning if the key in the configuration dict has changed. Here the value is changed.

Reply:
oh, yes, sorry. missed that